packages/story-editor/src/app/story/effects/useLoadStory.js

```diff
@@ -28,134 +28,130 @@ import { useHistory } from '../../history';
-          capabilities = {
-            publish: false,
-            'assign-author': false,
-          },
-          lock_user: lockUser,
-          featured_media: featuredMedia,
-          publisher_logo: publisherLogo,
-          taxonomies,
-          terms,
-        } = post;
-
-        const [prefix, suffix] = permalinkTemplate.split(
-          /%(?:postname|pagename)%/
-        );
-        // If either of these is undefined, the placeholder was not found in settings.
-        const foundSettings = prefix !== undefined && suffix !== undefined;
-        const permalinkConfig = foundSettings
-          ? {
-              prefix,
-              suffix,
-            }
-          : null;
-        const statusFormat = status === 'auto-draft' ? 'draft' : status;
-
-        // First clear history completely.
-        clearHistory();
-
-        // If there are no pages, create empty page.
-        const storyData =
-          storyDataRaw && migrate(storyDataRaw, storyDataRaw.version || 0);
-        const pages =
-          storyData?.pages?.length > 0 ? storyData.pages : [createPage()];
-
-        // Initialize color/style presets, if missing.
-        // Otherwise ensure the saved presets are unique.
-        if (!globalStoryStyles.colors) {
-          globalStoryStyles.colors = [];
-        } else {
-          globalStoryStyles.colors = getUniquePresets(globalStoryStyles.colors);
-        }
-        if (!globalStoryStyles.textStyles) {
-          globalStoryStyles.textStyles = [];
-        } else {
-          globalStoryStyles.textStyles = getUniquePresets(
-            globalStoryStyles.textStyles
-          );
-        }
-
-        // Set story-global variables.
-        const story = {
-          storyId,
-          title,
-          status: statusFormat,
-          author,
-          date,
-          modified,
-          excerpt,
-          slug,
-          link,
-          lockUser,
-          featuredMedia,
-          permalinkConfig,
-          publisherLogo,
-          password,
-          previewLink,
-          editLink,
-          embedPostLink,
-          currentStoryStyles: {
-            colors: storyData?.currentStoryStyles?.colors
-              ? getUniquePresets(storyData.currentStoryStyles.colors)
-              : [],
-          },
-          globalStoryStyles,
-          autoAdvance: storyData?.autoAdvance,
-          defaultPageDuration: storyData?.defaultPageDuration,
-          backgroundAudio: storyData?.backgroundAudio,
-          taxonomies,
-          terms,
-        };
-
-        // TODO read current page and selection from deeplink?
-        restore({
-          pages,
-          story,
-          selection: [],
-          current: null, // will be set to first page by `restore`
-          capabilities,
+      if (story) {
+        loadStory(storyId, story, restore, clearHistory);
+      } else {
+        getStoryById(storyId).then((post) => {
+          loadStory(storyId, post, restore, clearHistory);
```

Comment:
So we need to do this if no story was provided via `initialEdits`? When would that be the case?

Reply:
In the context of WordPress, we are only providing the demo story via `initialEdits` ( playground can also use `initialEdits` to load story ). Regular stories are still being loaded via `getStoryById` using the preloaded API data in. the same way.